services/workflows-service/src/business-report/business-report.controller.internal.ts

```diff
@@ -21,14 +22,38 @@ export class BusinessReportControllerInternal {
   @swagger.ApiForbiddenResponse({ type: errors.ForbiddenException })
   async getLatestBusinessReport(
     @CurrentProject() currentProjectId: TProjectId,
-    @Query() searchQueryParams: GetBusinessReportsDto,
+    @Query() searchQueryParams: GetBusinessReportDto,
   ) {
     return await this.businessReportService.findFirst(
       {
         where: {
           businessId: searchQueryParams.businessId,
           type: searchQueryParams.type,
         },
+        orderBy: {
+          createdAt: 'desc',
+        },
+      },
+      [currentProjectId],
+    );
+  }
+
+  @common.Get()
+  @swagger.ApiOkResponse({ type: [String] })
+  @swagger.ApiForbiddenResponse({ type: errors.ForbiddenException })
+  async getBusinessReports(
+    @CurrentProject() currentProjectId: TProjectId,
+    @Query() searchQueryParams: GetBusinessReportsDto,
+  ) {
+    return await this.businessReportService.findMany(
+      {
+        where: {
+          businessId: searchQueryParams.businessId,
+          ...(searchQueryParams.type ? { type: searchQueryParams.type } : {}),
```

Comment:
you can have it with undefied value its like not have a condition 
```js
type: searchQueryParams?.type
```

Reply:
I don't see any preference or additional values for that